common/datavalues2/src/columns/series.rs

```diff
@@ -31,6 +31,25 @@ use crate::StringColumn;
 pub struct Series;
 
 impl Series {
+    /// Type promotion to identify the nullable value.
+    /// Type promote to the nearest higher type.
+    /// UInt8 -> UInt16 -> UInt32 -> UInt64 -> bitmap<128>
+    /// Int8 -> Int16 -> Int32 -> Int64 -> bitmap<128>
+    pub(crate) fn _type_promotion(column: &ColumnRef, _datatype: TypeID) -> Option<TypeID> {
```

Comment:
We already have `construct_numeric_type` && `numeric_byte_size`.

Reply:
> We already have `construct_numeric_type` && `numeric_byte_size`.
oops, duplicated ..

Reply:
I think the construct_numeric_type method is too complex for type promotion,because we do not care about the 'sign/unsign, floating/nofloating' thing, we need only make the DataType promoted, and get the null positioned in the extra bits allocated. @sundy-li